src/codegen/codegen.go

```diff
@@ -0,0 +1,43 @@
+package codegen
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/thought-machine/please/src/core"
+)
+
+func UpdateGitignore(graph *core.BuildGraph, labels []core.BuildLabel, gitignore string) error {
```

Comment:
think this should probably come from the `scm` package and be abstracted by that?